activiti-cloud-services-runtime-bundle/activiti-cloud-services-subscriptions/src/main/java/org/activiti/services/subscriptions/behavior/BroadcastSignalEventActivityBehavior.java

```diff
@@ -0,0 +1,45 @@
+package org.activiti.services.subscriptions.behavior;
+
+import org.activiti.bpmn.model.Signal;
+import org.activiti.bpmn.model.SignalEventDefinition;
+import org.activiti.cloud.services.api.commands.SendSignalCmd;
+import org.activiti.engine.delegate.DelegateExecution;
+import org.activiti.engine.delegate.Expression;
+import org.activiti.engine.impl.bpmn.behavior.IntermediateThrowSignalEventActivityBehavior;
+import org.activiti.engine.impl.context.Context;
+import org.activiti.engine.impl.interceptor.CommandContext;
+import org.activiti.engine.impl.persistence.entity.ExecutionEntity;
+import org.springframework.context.ApplicationEventPublisher;
+
+public class BroadcastSignalEventActivityBehavior extends IntermediateThrowSignalEventActivityBehavior {
+    
+    private static final long serialVersionUID = 1L;
+
+    protected final SignalEventDefinition signalEventDefinition;
+
+    private final ApplicationEventPublisher eventPublisher;
+
+    public BroadcastSignalEventActivityBehavior(ApplicationEventPublisher eventPublisher, SignalEventDefinition signalEventDefinition,
+                                                        Signal signal) {
+        super(signalEventDefinition, signal);
+        this.eventPublisher = eventPublisher;
+        this.signalEventDefinition = signalEventDefinition;
+    }
+
+    public void execute(DelegateExecution execution) {
+
+        CommandContext commandContext = Context.getCommandContext();
```

Comment:
This extended behaviour should: 
1) first call super() behaviour
2) publish the event